pyam/core.py

```diff
@@ -1729,6 +1730,126 @@ def col_apply(self, col, func, *args, **kwargs):
         else:
             self.meta[col] = self.meta[col].apply(func, *args, **kwargs)
 
+    def add(self, a, b, name, axis="variable", append=False):
+        """Compute the addition of timeseries data between `a` and `b` along an `axis`
+
+        This function computes `a + b`. If `a` or `b` are lists, the method applies
+        :meth:`pandas.groupby().sum() <pandas.core.groupby.GroupBy.sum>` on each group.
+        If either `a` or `b` are not defined for a row, no value is computed
+        for that row.
+
+        Parameters
+        ----------
+        a, b : str or list of str
+            Items to be used for the addition.
+        name : str
+            Name of the computed timeseries data on the `axis`.
+        axis : str, optional
+            Axis along which to compute.
+        append : bool, optional
+            Whether to append aggregated timeseries data to this instance.
+
+        Returns
+        -------
+        :class:`IamDataFrame` or **None**
+            Computed timeseries data or None if `append=True`.
+        """
+        _value = _op_data(self, (a, b), name, "add", axis=axis)
+        if append:
+            self.append(_value, inplace=True)
+        else:
+            return IamDataFrame(_value, meta=self.meta)
```

Comment:
I guess it is possible some models/scenarios could be dropped in some of the operations if data is missing. should this be 

```
return IamDataFrame(_value, meta=self.meta[_value.index])
```

?

Reply:
and if so - then this change should propagate down

Note, btw, this is why I made originally a generic `binary_op()` function, which did all the set up/tear down work - but can also be updated later!

Reply:
This is already happening automatically during initialization... https://github.com/IAMconsortium/pyam/blob/d4eafc319480b8a96bba3f6cc500c89f59646cf6/pyam/core.py#L189